dlgr/griduniverse/experiment.py

```diff
@@ -1585,6 +1636,151 @@ def handle_build_wall(self, msg):
+            error_msg = {
+                "type": "consume_error",
+                "player_id": player.id,
+                "player_item": player_item.serialize(),
+            }
+            self.publish(error_msg)
+            return
+
+        player_item.remaining_uses -= 1
+        if not player_item.remaining_uses:
+            self.grid.items_consumed.append(player_item)
+            player.current_item = None
+
+        if player.color_idx > 0:
+            calories = player_item.calories
+        else:
+            calories = player_item.calories * self.relative_deprivation
+
+        player.score += calories
+        if player_item.public_good:
+            for player_to in self.grid.players.values():
+                player_to.score += player_item.public_good
+
+    def handle_item_pick_up(self, msg):
+        player = self.grid.players[msg["player_id"]]
+        player_item = player.current_item
+        position = tuple(msg["position"])
+        location_item = self.grid.item_locations.get(position)
+        if player_item is not None or location_item is None:
+            error_msg = {
+                "type": "action_error",
+                "player_id": player.id,
+                "position": list(position),
+                "item": location_item and location_item.serialize(),
+                "player_item": player_item and player_item.serialize(),
+            }
+            self.publish(error_msg)
+            return
+        location_item.position = None
+        del self.grid.item_locations[position]
+        player.current_item = location_item
+        self.grid.items_updated = True
+
+    def handle_item_transition(self, msg):
+        player = self.grid.players[msg["player_id"]]
+        player_item = player.current_item
+        position = tuple(msg["position"])
+        location_item = self.grid.item_locations.get(position)
+        transition = None
+
+        # If the target item has only 1 remaining use, then we try to find a
+        # `last_use` transition
+        if location_item and location_item.remaining_uses == 1:
+            transition = self.transition_config.get(
+                ("last", player_item and player_item.item_id, location_item.item_id)
+            )
+
+        if transition is None:
+            transition = self.transition_config.get(
+                (
+                    player_item and player_item.item_id,
+                    location_item and location_item.item_id,
+                )
+            )
+
+        if transition is None:
+            error_msg = {
+                "type": "action_error",
+                "player_id": player.id,
+                "position": list(position),
+                "item": location_item and location_item.serialize(),
+                "player_item": player_item and player_item.serialize(),
+            }
+            self.publish(error_msg)
+            return
+
+        modify_actor, modify_target = transition.get("modify_uses", (0, 0))
```

Comment:
```suggestion
        # these values may be positive or negative, so we may add or remove uses
        modify_actor, modify_target = transition.get("modify_uses", (0, 0))
```

(Not sure this is true, but I find it confusing so some comment helpful). Also, `modify_x` sounds like a function name, which threw me for a second. `difference_actor` & `difference_target`? I can't think of anything great atm.

Reply:
It could indeed be positive or negative, so I like this comment. This comes from the `modify_uses` property that Natalia suggested, and I think `modify` makes sense because we're modifying the `remaining_uses` count, but maybe these should be more explicitly `modify_actor_uses` an `modify_target_uses`?

Reply:
I've added your comment and renamed the variables to be more clear.